styles/AdvancedEditor.css

```diff
@@ -119,26 +120,26 @@ Next levels are shifted 'up' by 1
 
 .ProseMirror
   > div.body
-  > section
-  > section
-  > section:not(.toc)
+  section
+  section
+  section:not(.toc)
   > .block-section_title
   > h1.empty-node[data-placeholder]::before {
   content: counter(level_1, decimal) '.' counter(level_2, decimal) '.'
     counter(level_3, decimal) '. ' attr(data-placeholder);
 }
 /* LVL 5 */
-.ProseMirror > div.body > section > section > section > section:not(.toc) {
+.ProseMirror > div.body section section section section:not(.toc) {
   counter-increment: level_4;
   counter-reset: level_5;
 }
 
 .ProseMirror
   > div
-  > section
-  > section
-  > section
-  > section
+  section
+  section
+  section
+  section
```

Comment:
I agree with @mnatsheh that we are here pushing the limit of what can be achieved with css. Selectors a quite complicated and a plugin should provide a more concise and systematical solution

Reply:
@fwaisi , the plugin would not update any children, it would just keep track of the label of each section, and the section_title view would use this info. See for example the affilitations plugin. There is also a ticket (LEAN-3195) to have these numbers appear in the outline, so this info will be useful elsewhere too.

Generally speaking, the purpose of plugins is exactly storing data we don't want in the doc (or in JATS).

Reply:
As far as I remember from the video on the ticket, a part of the problem is that the numbering is inconsistent with track-changes status and if we need to take into account track changes when numbering, plugin will allow more control over it.

Reply:
@mbartenev-atypon, @mnatsheh, sure I will implement the plugin.